src/scenes/Blog/components/Previews/components/List/List.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import Card from '../Card';
+
+import styles from './List.module.scss';
+
+const STEP_VALUE = 6;
+
+const List = ({ items }) => {
+  const [numberOfRendered, setNumberOfRendered] = useState(STEP_VALUE);
+
+  const handleClick = () => {
+    const value = numberOfRendered + STEP_VALUE;
+    setNumberOfRendered(value);
+  };
+
+  return (
+    <div className={styles.container}>
+      <ul className={styles.list}>
+        {items.map((item, index) => {
+          const inlineStyles = { animationDelay: `0.${index}s` };
+
+          return numberOfRendered > index ? (
```

Comment:
we can change it to for loop to not iterate on elements that won't be shown:
```
const blogPosts = [];

  for (let i = 0; i <= numberOfRendered; i++) {
    const inlineStyles = { animationDelay: `0.${i}s` };

    blogPosts.push(
      <li key={items[i].id} style={inlineStyles} className={styles.item}>
        <Card {...items[i]} />
      </li>
    );
  }

  return (
    <div className={styles.container}>
      <ul className={styles.list}>{blogPosts}</ul>

      {numberOfRendered < items.length ? (
        <div className={styles.buttonWrapper}>
          <button className={styles.button} onClick={handleClick}>
            Show More
          </button>
        </div>
      ) : null}
    </div>
  );```

Reply:
I like it.